src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -25,6 +25,17 @@ public partial class Ping : IDisposable
         private const int Disposed = 2;
         private int _status = Free;
 
+        public Ping()
+        {
+            if (GetType() == typeof(Ping)) // Not derived class. No use of finalizer here, suppress it immediately
+                GC.SuppressFinalize(this);
```

Comment:
Nit: maybe we can add more comments on why this is done in the ctor. 

I now understand that since all internal resources are already finalizable types (SafeHandles mostly), Ping doesn't have anything to release. Still, calling this function in the ctor is not commonly seen or recommended by MSDN:

https://msdn.microsoft.com/en-us/library/b1yfkh5e(v=vs.110).aspx#finalizable_types